nav2_util/include/nav2_util/simple_action_server.hpp

```diff
@@ -103,9 +114,25 @@ class SimpleActionServer
       action_name_,
       std::bind(&SimpleActionServer::handle_goal, this, _1, _2),
       std::bind(&SimpleActionServer::handle_cancel, this, _1),
-      std::bind(&SimpleActionServer::handle_accepted, this, _1));
+      std::bind(&SimpleActionServer::handle_accepted, this, _1),
+      rcl_action_server_get_default_options(),
+      callback_group_);
+    if (spin_thread_) {
+      auto executor = std::make_shared<rclcpp::executors::SingleThreadedExecutor>();
```

Comment:
Please make the executor a class variable. I can foresee a time when we want to be able to access this within the action server's execution or make a `getExecutor()` method to mess with things during runtime. While its not strictly required today, it seems like a bad idea to have a thing running for which the program cannot access its executor after initialization.